whereby/browser-sdk
# Upgrade Storybook

* Upgrade version
* Migrate Storybook main file
* Migrate package.json's storybook scripts
* Create babel config as SB6 had it, but SB7 does not have it, however it needs it.
* Lock devDeps' deps as storybook + yarn classic (v1) cannot install them properly. See: https://github.com/storybookjs/storybook/issues/22431#issuecomment-1630086092

### Test plan

1. Install deps (you may need to `rm -rf node_modules`)
2. Run storybook with `yarn storybook`
3. Make sure it starts and the components are visible
4. Build the sdk and add it to your sample app => everything should work as before

### Yarn audit

Run `yarn audit` in this browser-sdk repo.

| Branch | Vulnerabilities found | Packages audited | Severity: Moderate | Severity: High | Severity: Critical |
| -- | -- | -- | -- | -- | -- | 
| development | 55 | 2200 | 46 | 9 | 0 |
| current | 0 | 1380 | 0 | 0 | 0 |

### Gotchas

The devDeps' dep locks could be removed when we upgrade yarn from v1 to v3. But yarn v3 needs minimum Node.js v18 which is not available everywhere. Hopefully we can get there soon.
